devfile.yaml

```diff
@@ -16,6 +16,7 @@ metadata:
 parent:
   id: go
   registryUrl: 'https://registry.devfile.io'
+  version: 2.1.0
```

Comment:
- Sounds reasonable to me considering the commands are present in the parent. But we also have the same components just different names, they sound like duplicates to me too
- One question we should discuss, when a devfile is being parsed, both the main and parent devfile will be _**flattened**_. Now we will have two deploy.yaml, one from stack and one from this repository, how are they going to be used?

Reply:
The end goal is, we will have one devfile, there will be no stack and sample, there will be only stack devfile.. So i can see why the stack devfile have the logic from the sample devfile but since we are not there yet, we should discuss how we want to handle this!

Reply:
@maysunfaisal 

I've tested some cases in order to dig a little bit on the behavior for the `deploy.yaml` file.

* I've tried to remove the `deploy.yaml` and the `components` inside the `devfile.yaml` file. The result I had from the component detection query was mentioning that no `deploy.yaml` was found inside the `kubernetes` directory. The log:
```
..."error":"failed to convert kubernetes uri to inlined for component 'deploy': error getting kubernetes resources definition information: failed to retrieve https://raw.githubusercontent.com/thepetk/devfile-sample-go-basic/test_devfile/kubernetes/deploy.yaml, 404: Not Found","errorVerbose":"failed to retrieve https://raw.githubusercontent.com/thepetk/devfile-sample-go-basic/test_devfile/kubernetes/deploy.yaml, 404: Not Found\ngithub.com/devfile/library/v2...
```

* When I've moved the `deploy.yaml` of the sample into a kubernetes dir (check branch [test_devfile](https://github.com/thepetk/devfile-sample-go-basic/tree/test_devfile) of my fork) the app was able to detect the deploy yaml and do a successful build for a devfile without components (inherited from parent).

So as far as I understand the current behavior is:
- If a parent or a child devfile are mentioning a `deploy.yaml` we are expecting to find it inside the `kubernetes/` dir and this file is not inherited.

> But we also have the same components just different names, they sound like duplicates to me too

Yes and if we use the parent's ones we are still fine.

> So i can see why the stack devfile has the logic from the sample devfile but since we are not there yet, we should discuss how we want to handle this!

Definitely I agree. I'll follow up once I have more info.

Reply:
@maysunfaisal as it was discussed I've added all `commands` and `components` under the `parent` block to override the parent ones for now. Ofc in the future when we will not have to use two different `devfiles` and `deploys` we can change again the file.